src/Data/BigInt.purs

```diff
@@ -111,6 +115,18 @@ foreign import shl :: BigInt -> Number -> BigInt
 -- | Shift the bits right and maintain pos/neg.
 foreign import shr :: BigInt -> Number -> BigInt
 
+-- Note: this function should not be exported!
+-- It's only safe if used with type-level integers.
+foreign import fromTypeLevelInt :: String -> BigInt
```

Comment:
What about renaming this `fromTypeLevelIntImpl` and then having `fromTypeLevelInt` be the exported name? It's not immediately clear to me what `TL` means in the function name.